packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreAliased.spec.ts

```diff
@@ -0,0 +1,119 @@
+describeFullCompat("GC Data Store Aliased", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    const dataObjectFactory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: {
+            disableSummaries: true,
+        },
+        gcOptions: {
+            gcAllowed: true,
+        },
+    };
+    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+        runtime.IFluidHandleContext.resolveHandle(request);
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        [innerRequestHandler],
+        runtimeOptions,
+    );
+    
+    // Enable config provider setting to write GC data at the root.
+    const settings = { "Fluid.GarbageCollection.LogUnknownOutboundRoutes": "true" };
+    const configProvider = mockConfigProvider(settings);
+
+    let container1: IContainer;
+    let container2: IContainer;
+    let mainDataStore1: TestDataObject;
+    let mainDataStore2: TestDataObject;
+
+     async function summarizeOnContainer(container: IContainer) {
+        const dataStore = await requestFluidObject<TestDataObject>(container, "default");
+        return dataStore.containerRuntime.summarize({runGC: true, trackState: false});
+    }
+
+    const createContainer = async (): Promise<IContainer> => {
+        return provider.createContainer(runtimeFactory, { configProvider });
+    }
+    const loadContainer = async (): Promise<IContainer> => {
+        return provider.loadContainer(runtimeFactory, { configProvider });
+    };
+    beforeEach(async () => {
+        provider = getTestObjectProvider();
+
+        // Create a Container for the first client.
+        container1 = await createContainer();
+        container2 = await loadContainer();
+
+        mainDataStore1 = await requestFluidObject<TestDataObject>(container1, "default");
+        mainDataStore2 = await requestFluidObject<TestDataObject>(container2, "default");
+        await provider.ensureSynchronized();
+    });
+
+    // TODO: fully validate that GC is notified. Currently this tests the race condition
+    // where a remote datastore is summarized before the alias op arrives when trySetAlias is called.
+    itExpects("GC is notified when datastores are aliased.", [], async () => {
+        await summarizeOnContainer(container2);
+        const containerRuntime1 = mainDataStore1.containerRuntime;
+        const aliasableDataStore1 = await containerRuntime1.createDataStore("TestDataObject");
+
+        (aliasableDataStore1 as any).fluidDataStoreChannel.bindToContext();
+        await provider.ensureSynchronized();
+
+        // Summarize before aliasing
+        const containerRuntime2 = mainDataStore2.containerRuntime;
```

Comment:
nit: move this to where it is used. It's a good idea to initialize / declare variables as and when needed.
Same comment for the other test.